### ISSUE TITLE
Owners and Editors are not deleted from a list when removed from a data_source

### ISSUE THREAD

**Issue by dpc22** [sympa-community/sympa#1315]
Version
-------

6.2.66

Installation method
-------------------

My own rpm, derived from "official" RHEL 7 rpm

Expected behavior
-----------------

If a list defines owners or editors/moderators using an external data source then when an email address is removed from that data source it should be removed from the user_admin table at the next synchronisation event.

Actual behavior
---------------

This is not the case. Sympa _claims_ to have deleted the email addresses on each synchronisation event:
```
Jan  6 15:46:09 lists-1 wwsympa[59465]: info Sympa::Request::Handler::include::_twist() Sympa::Request <action=include;context=ast-test@lists.cam.ac.uk;role=editor>: Success, 0 added, 3 deleted, 1 updated
Jan  6 15:48:00 lists-1 wwsympa[59466]: info Sympa::Request::Handler::include::_twist() Sympa::Request <action=include;context=ast-test@lists.cam.ac.uk;role=editor>: Success, 0 added, 3 deleted, 1 updated
Jan  6 15:48:14 lists-1 wwsympa[59462]: info Sympa::Request::Handler::include::_twist() Sympa::Request <action=include;context=ast-test@lists.cam.ac.uk;role=editor>: Success, 0 added, 3 deleted, 1 updated
```
but this doesn't actually take effect, so you see "3 deleted" every time above.

Additional information
----------------------

I took a look at the code, and I'm pretty sure that this is a trivial bug in: ./lib/Sympa/Request/Handler/include.pm

```
sub _expire_users {
    my $list            = shift;
    my $role            = shift;
[...]
        if ($role eq 'member') {
            $list->delete_list_member(users => \@emails);
        } else {
            $list->delete_list_admin($role, \@emails);
        }
```
$list->delete_list_admin() expects a list rather than a list reference. If you give it a list reference it behaves as if you gave it an empty list.

The obvious change:
```
--- ./lib/Sympa/Request/Handler/include.pm-DIST 2021-08-30 05:53:42.000000000 +0100
+++ ./lib/Sympa/Request/Handler/include.pm      2022-01-07 07:10:49.546937920 +0000
@@ -639,7 +639,7 @@
         if ($role eq 'member') {
             $list->delete_list_member(users => \@emails);
         } else {
-            $list->delete_list_admin($role, \@emails);
+            $list->delete_list_admin($role, @emails);
         }
 
         foreach my $email (@emails) {
```
fixes the problem for me.

**Comment by ikedas** [comment]
Hi @dpc22, thank you for reporting bug.
This bug seems fixed with commit 28e5885 .  Please check it if possible.

**Comment by ikedas** [comment]
Duplicate of #1272 .
